staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1597,6 +1597,30 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		return e.KeyFunc(genericapirequest.NewContext(), accessor.GetName())
 	}
 
+	// reverse key into namespace and name
+	// Examples:
+	//		key:    /registry/pods/namespace/name
+	//		prefix: /pods
+	reverseKeyFunc := func(key string) (name string, namespace string, err error) {
```

Comment:
This can be normal function, not an anonymous one. This would allow us to unit test with `NoNamespaceKeyFunc` and `NamespaceKeyRootFunc` like `reverseKeyFunc(NoNamespaceKeyFunc(x)) == x`